dataflow/src/main/java/com/example/spanner_migration/SpannerStreamingWrite.java

```diff
@@ -206,27 +202,32 @@ public static void main(String[] args) {
         .apply("CU->Mutations", ParDo.of(new UpdateMutations(options.getTable())));
     // create mutations for deletes
     PCollection<Mutation> deletes = messages.apply("Delete?", ParDo.of(new DeleteItems()))
-        .apply("D->Mutations", ParDo.of(new DeleteMutations(options.getTable())));
+        .apply("D->Mutations",
+            ParDo.of(new DeleteMutations(options.getTable())));
 
     // Merge both sets of mutations
     PCollectionList<Mutation> merged = PCollectionList.of(updates).and(deletes);
 
     // Create fixed windows on unbounded (pub/sub) source
     PCollection<Mutation> mergedWindowed = merged.apply("Merging Mutations",
             Flatten.<Mutation>pCollections())
-        .apply("Creating Windows", Window.<Mutation>into(
-            FixedWindows.of(Duration.standardSeconds(Long.parseLong(options.getWindow())))));
+        .apply("Creating Windows", Window
+            .<Mutation>into(
+                FixedWindows.of(Duration.standardSeconds(Long.parseLong(options.getWindow()))))
+            .triggering(
+                AfterProcessingTime.pastFirstElementInPane()
+                    .plusDelayOf(Duration.standardSeconds(10)))
+            .withAllowedLateness(Duration.standardMinutes(30)).discardingFiredPanes());
```

Comment:
I propose that either that **withAllowedLateness** value be increased, or a note be added to guide / code that calls out that this value may be increased so that there is sufficient time to run the batch portion of the migration, _before_ the streaming portion is started.